WordPress/Classes/System/WindowManager.swift

```diff
@@ -45,20 +45,25 @@ class WindowManager: NSObject {
         showSignInUI()
     }
 
-    func dismissFullscreenSignIn(completion: Completion? = nil) {
+    func dismissFullscreenSignIn(with blog: Blog? = nil, completion: Completion? = nil) {
         guard isShowingFullscreenSignIn == true && AccountHelper.isLoggedIn == true else {
             return
         }
 
-        showAppUI(completion: completion)
+        showAppUI(for: blog, completion: completion)
     }
 
     /// Shows the UI for authenticated users.
     ///
-    func showAppUI(completion: Completion? = nil) {
+    func showAppUI(for blog: Blog? = nil, completion: Completion? = nil) {
         isShowingFullscreenSignIn = false
-
         show(WPTabBarController.sharedInstance(), completion: completion)
+
+        guard let blog = blog else {
+            return
+        }
+
+        WPTabBarController.sharedInstance()?.mySitesCoordinator.showBlogDetails(for: blog)
```

Comment:
What do you think about creating a method such as `switchCurrentlyBlog` or `switchBlog` on `WPTabBarController`?

That way we avoid using the `mySitesCoordinator` here.

Reply:
Fixed! 87c8559